src/tcop/tcop.cpp

```diff
@@ -133,14 +126,255 @@ ResultType TrafficCop::ExecuteStatement(
     const std::string &query, std::vector<StatementResult> &result,
     std::vector<FieldInfo> &tuple_descriptor, int &rows_changed,
     std::string &error_message, const size_t thread_id UNUSED_ATTRIBUTE) {
+
   LOG_TRACE("Received %s", query.c_str());
 
-  // Prepare the statement
   std::string unnamed_statement = "unnamed";
   auto statement = PrepareStatement(unnamed_statement, query, error_message);
+  // After prepareStatement, statements that might be aborted in prepareStatement will be filtered out.
+  // for example, create a table but that table exists; drop a table but that table has been dropped
+  if (statement.get() == nullptr) {
+    if (single_statement_txn_) {
+      LOG_TRACE("SINGLE ABORT!");
+      rows_changed = 0;
```

Comment:
I totally agree with you. Some test cases are not unified, and I also want to change it. But now my urgent assignment is to rewrite packet_manager and I must modify many functions from packet_manager and tcop which means I need to change many test cases again.